### PR TITLE
:sparkles: Kai Server Refactor

### PR DESCRIPTION
**Pull request by pranavgaikwad**
- Remove KaiApplication 
  - Removes all APIs to get solutions for a file
- Web Server starts with IncidentStore instead of KaiApplication
  - IncidentStore can be inited from a config
- IncidentStore Changes
  - load_report() no longer calls post_process() on producers
  - new post_process() function exposed on IncidentStore that needs to be explicitely called to run producer post_process()
- HubImporter Changes
  - Added new method to the main loop that runs post_process() periodically for a small subset of solutions

### CODE REVIEW

**Comment by pranavgaikwad** [comment]
Hey folks, I may have broke some things with this change, things that I know of:

1. run_demo.py: I need to work with @JonahSussman to restore this back with Kai client

Please comment if you know anything else that may be affected.

File: kai/routes/get_solutions.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    solutions = request.app[
+        # TODO (pgaikwad): type hint this
+        "kai_incident_store"
+        # web.AppKey("kai_incident_store", IncidentStore)
+    ].find_solutions(**params.model_dump())
+
+    return web.json_response(
+        {
+            "solutions": [sol.model_dump_json() for sol in solutions],
```
Comment (shawn-hurley) [inline kai/routes/get_solutions.py:38]: What do folks think of making this a specific dataclass?

**Review by shawn-hurley — APPROVED** [review]
Other than data classing the result for the JSON responses.

**Comment by pranavgaikwad** [comment]
Hey folks, I have disabled run_demo and evaluation tests because they are broken. I will get run_demo in a followup.

**Pull request merged**